pkg/rtc/transport.go

```diff
@@ -306,6 +335,96 @@ func (t *PCTransport) createAndSendOffer(options *webrtc.OfferOptions) error {
 	return nil
 }
 
+func (t *PCTransport) preparePC(previousAnswer webrtc.SessionDescription) error {
+	// sticky data channel to first m-lines, if someday we don't send sdp without media streams to
+	// client's subscribe pc after joining, should change this step
+	parsed, err := previousAnswer.Unmarshal()
+	if err != nil {
+		return err
+	}
+	fp, fpHahs, err := extractFingerprint(parsed)
+	if err != nil {
+		return err
+	}
+
+	offer, err := t.pc.CreateOffer(nil)
+	if err != nil {
+		return err
+	}
+
+	t.pc.SetLocalDescription(offer)
+
+	pc2, err := webrtc.NewPeerConnection(webrtc.Configuration{})
+	if err != nil {
+		return err
+	}
+	defer pc2.Close()
+
+	pc2.SetRemoteDescription(offer)
+	ans, err := pc2.CreateAnswer(nil)
+	if err != nil {
+		return err
+	}
+	parsedAns, _ := ans.Unmarshal()
+	fpLine := fpHahs + " " + fp
+	replaceFP := func(attrs []sdp.Attribute, fpLine string) {
+		for k := range attrs {
+			if attrs[k].Key == "fingerprint" {
+				attrs[k].Value = fpLine
+			}
+		}
+
+	}
+	replaceFP(parsedAns.Attributes, fpLine)
+	for _, m := range parsedAns.MediaDescriptions {
+		replaceFP(m.Attributes, fpLine)
+	}
+	bytes, err := parsedAns.Marshal()
+	if err != nil {
+		return err
+	}
+	ans.SDP = string(bytes)
+
+	return t.pc.SetRemoteDescription(ans)
+}
+
+func (t *PCTransport) initPCWithPreviousOffer(previousOffer webrtc.SessionDescription) error {
```

Comment:
Looks like we are using previous answer from client on subscriber PC to re-create the previous offer. So, I am getting confused with the function name.

Reply:
Thank you , first I want client send previous offer and later use answer for it, will change it